README.md

```diff
@@ -1,240 +1,226 @@
+  > Note: x.x.x-x is the version of the docker provider which is determined from version info in version file
+
+### Build Docker Image
+
+1.  Navigate to below directory to build the docker image
+  ```
+  cd ~/Docker-Provider/kubernetes/linux/
+  ```
+2. Upload docker-cimprov-x.x.x-x.universal.x86_64.sh from ~/Docker-Provider/target/Linux_ULINUX_1.0_x64_64_Release/ to azure blob storage account blob
+
+ ```
+ AZURE_SUBSCRIPTIONID=<azure-subscription-id> # subscriptionId for the azure storage account (new or exist)
+ STORAGE_ACCOUNT_RG=<resource-group>
+ STORAGE_ACCOUNT_LOCATION=<location>
+ STORAGE_ACCOUNT_NAME=<account-name>
+ STORAGE_ACCOUNT_BLOB_NAME=<blob-name> # for example agentshellbundle
+
+ # login to azure interactively and set the subscription id
+ az login --use-device-code
+ az account set -s $AZURE_SUBSCRIPTIONID
+
+ # create rg and storage account if this doesnt exist one already
+ az group create -n $STORAGE_ACCOUNT_RG -l $STORAGE_ACCOUNT_LOCATION
+ az storage account create -n $STORAGE_ACCOUNT_NAME -g $STORAGE_ACCOUNT_RG -l $STORAGE_ACCOUNT_LOCATION
+
+ # create blob container
+ az storage container create --account-name $STORAGE_ACCOUNT_NAME --name $STORAGE_ACCOUNT_BLOB_NAME --auth-mode login --public-access container
+
+ # upload docker-cimprov shell bundle to storage account. please specify the correct version
+ az storage blob upload --account-name $STORAGE_ACCOUNT_NAME --container-name $STORAGE_ACCOUNT_BLOB_NAME --name docker-cimprov-10.0.0-0.universal.x86_64.sh --file  ~/Docker-Provider/target/Linux_ULINUX_1.0_x64_64_Release/docker-cimprov-10.0.0-0.universal.x86_64.sh
+ # replace the placeholders in this url
+ https://$STORAGE_ACCOUNT_NAME.blob.core.windows.net/$STORAGE_ACCOUNT_BLOB_NAME/docker-cimprov-10.0.0-0.universal.x86_64.sh
+
+ # verify this url valid and exist by running the curl --head command as below
+ curl --head https://mydevomsagentsa.blob.core.windows.net/agentshellbundle/docker-cimprov-10.0.0-0.universal.x86_64.sh
+ ```
+  > Note: x.x.x-x is the version of the docker provider which is determined from version info in docker.version file
+3. Update the azure storage blob location of docker-cimprov-x.x.x-x.universal.x86_64.sh in setup.sh
+ ```
+ # update the setup.sh in this with the url in above step in place of  https://github.com/microsoft/Docker-Provider/releases/download/..
+ cd ~/Docker-Provider/kubernetes/linux/
+ ```
+4. Build the Docker image via below command
+```
+   docker build -t  <repo>/<imagename>:<imagetag> .
+```
+5. Push the Docker image to docker repo
+```
+   docker push  <repo>/<imagename>:<imagetag>
+```
+
+## Windows Agent
+> Note: To build the Windows Agent Image, you will need Windows 10 Pro or higher machine with Docker for Windows
+### Install Pre-requisites
+1. Install .Net Core SDK 2.2 or higher from https://dotnet.microsoft.com/download if you dont have installed already
+2. Install go1.14.1 if you havent installed already
```

Comment:
we should update this to install pre-reqs using chocolatey later.

Reply:
> we should update this to install pre-reqs using chocolatey later.

> yes, this can be done and thought of doing that,  but challenge installing docker since docker installer can be downloaded after login to docker hub. do you know is there public msi location for docker installer which we can use ?

Reply:
Yes, Lets handle this later since install gcc exe on windows cant automated since this exe doesnt way any commandline options and it has to be installed with UI.